lib/git/lib.rb

```diff
@@ -738,6 +738,7 @@ def tag(name, *opts)
 
     def fetch(remote, opts)
       arr_opts = [remote]
+      arr_opts << opts[:ref] unless opts[:ref].nil?
       arr_opts << '--tags' if opts[:t] || opts[:tags]
       arr_opts << '--prune' if opts[:p] || opts[:prune]
```

Comment:
I don't like us mixing `unless foo.nil?` and `if foo` syntax like this.

Are there any strong reasons why we couldn't rephrase this as:

```ruby
arr_opts << opts[:ref] if opts[:ref]
```

The only semantic difference I can think of with these two forms is that the previous will let you pass in `false` as a ref, but... that doesn't really make much sense anyway.

Reply:
No problem! I like the use of `unless` to express the idea of `We're going to add this, unless this thing is nil`. Of course your suggestion is totally reasonable: `We're going to add this if there is something to add`. I updated the pr for the style changes 👍

Reply:
Thanks @taquitos. I agree that the unless pattern can make the code more readable, I just want things to be uniform.